src/compiler/scala/tools/nsc/settings/StandardScalaSettings.scala

```diff
@@ -45,12 +47,15 @@ trait StandardScalaSettings {
   val nowarn =         BooleanSetting ("-nowarn", "Generate no warnings.") withAbbreviation "--no-warnings"
   val optimise:        BooleanSetting // depends on post hook which mutates other settings
   val print =          BooleanSetting ("-print", "Print program with Scala-specific features removed.") withAbbreviation "--print"
-  val target =         ChoiceSettingForcedDefault ("-target", "target", "Target platform for object files. All JVM 1.5 - 1.7 targets are deprecated.",
-                          List("jvm-1.5", "jvm-1.6", "jvm-1.7", "jvm-1.8"), "jvm-1.8") withAbbreviation "--target"
+  val target =         ChoiceSetting  ("-target", "target", "Target platform for object files.", List("jvm-1.8", "jvm-9", "9"), "jvm-1.8") withPostSetHook fixupTarget _ withAbbreviation "--target"
```

Comment:
How abuot we just advertise `"1.8", "9", ...` here, and rig up this setting to strip off the `jvm-` part before the `contains` call in `ChoiceSetting`.

```
      case List(x) if choices contains x  => value = x ; SomeOfNil
```

```scala
 = ChoiceSetting  ("-target", "target", "Target platform for object files.", List("jvm-1.8", "jvm-9", "9"), "jvm-1.8").withPostSetHook(fixupTarget).withPreSetHook(_.stripPrefix("jvm-").withAbbreviation("--target")
```

(where `withPreSetHook` is a new method on `ChoiceSetting)

Reply:
Adding a meth todo `ChoiceSetting` from this class is apparently non-trivial.